_posts/2016-05-05-cargo-pillars.md

```diff
@@ -0,0 +1,489 @@
+
+While this policy works well (and in fact is the same policy that system package
+managers use), it doesn't always do exactly what people expect, especially when
+it comes to coordinating a major version bump across the ecosystem. (In many
+cases, it would be preferable for Cargo to hard-error than assume that a
+dependency on `0.2.x` is simply unrelated to another dependency on `0.3.x`.)
+
+This problem is especially pernicious when multiple major versions of the same
+package expose global symbols (using `#[no_mangle]` for example, or by including
+other statically linked C libraries).
+
+We have some thoughts on ways to improve Cargo to handle these cases better,
+including the ability for a package to more explicitly express when a dependency
+is used purely internally and is not shared through its public interface. Those
+packages could be more readily duplicated if needed, while dependencies that are
+used in a package's public interface must not be.
+
+You should expect to see more on this topic in the months ahead.
+
+## Workflow
+
+As we've seen, **Cargo is not just a dependency manager, but Rust's primary
+workflow tool**.
+
+This allows Rust to have a rich ecosystem of interconnected dependencies,
+eliminating the need for applications to manually manage large dependency
+graphs. Applications can benefit from a vibrant ecosystem of small packages that
+do one thing and do it well, and let Cargo handle the heavy lifting of keeping
+everything up to date and compiling correctly.
+
+Even a small crate like the `datetime` example we built has a few dependencies
+on small, targeted crates, and each of those crates has some dependencies of its
+own.
+
+By defining shared, well-known workflows, like "build", "test", "bench", "run",
+and "docs", Cargo provides Rust programmers with a way to think about what
+they're trying to accomplish at a high level, and not have to worry about what
+each of those workflows mean for indirect dependencies.
+
+This allows us to get closer to the holy grail of making those indirect
+dependency graphs "invisible", empowering individuals to do more on their hobby
+projects, small teams to do more on their products, and large teams to have a
+high degree of confidence in the output of their work.
+
+We're all very proud of the way that comes together in
+[Servo](https://github.com/servo/servo), the Rust-based web browser built using
+Cargo. The code that they have written as a small team is impressive, but even
+more impressive is how much of Servo is maintained and used by the wider Rust
+community.
+
+Concretely, this means that Servo's encoding library (and many components like
+it) is not a deeply nested part of Servo's main tree, but rather an
+[external library](https://crates.io/crates/encoding) that anyone in the
+ecosystem can use. This makes it possible for other Rust libraries, like web
+frameworks, to easily use the same encoding library used by a web browser,
```

Comment:
I love the term "grade" here along the lines of a "browser-grade" dependency or something like that. Perhaps that could get wordsmithed in to emphasize how cool it is to use a dependency vetted by an entire web browser?